### PR TITLE
Cleanup cirq.Circuit api docs.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Gives better rendering.

### CODE REVIEW

File: cirq-core/cirq/circuits/circuit.py [hunk 1]
```diff
@@ -754,10 +772,18 @@ def findall_operations_with_gate_type(
...
+        Returns: True if `cirq.is_measurement(self)` is True otherwise False.
+        """
         return protocols.is_measurement(self)
 
     def are_all_measurements_terminal(self) -> bool:
-        """Whether all measurement gates are at the end of the circuit."""
+        """Whether all measurement gates are at the end of the circuit.
+
+        Returns: bool that is True if all measurements have no gates
+            following them on their respective qubits.
```
Comment (viathor) [inline cirq-core/cirq/circuits/circuit.py:785]: Optional suggestion:

```
Returns: True iff no measurement is followed by a gate.
```

File: cirq-core/cirq/circuits/circuit.py [hunk 2]
```diff
@@ -799,7 +825,11 @@ def are_all_matches_terminal(self, predicate: Callable[['cirq.Operation'], bool]
         return True
 
     def are_any_measurements_terminal(self) -> bool:
-        """Whether any measurement gates are at the end of the circuit."""
+        """Whether any measurement gates are at the end of the circuit.
+
+        Returns: bool that is True if any measurements have no gates
+            following them on their respective qubits.
```
Comment (viathor) [inline cirq-core/cirq/circuits/circuit.py:831]: Optional suggestion:

```
Returns: True iff some measurements are not followed by a gate.
```

**Review by viathor — APPROVED** [review]
LGTM

File: cirq-core/cirq/circuits/circuit.py (hunk 1 above)
Comment (MichaelBroughton) [reply cirq-core/cirq/circuits/circuit.py]: DOne.

File: cirq-core/cirq/circuits/circuit.py (hunk 2 above)
Comment (MichaelBroughton) [reply cirq-core/cirq/circuits/circuit.py]: Done.

**Pull request merged**